src/rules/noMagicNumbersRule.ts

```diff
@@ -34,23 +45,36 @@ export class Rule extends Lint.Rules.AbstractRule {
             Magic numbers should be avoided as they often lack documentation.
             Forcing them to be stored in variables gives them implicit documentation.
         `,
-        optionsDescription: "A list of allowed numbers.",
+        optionsDescription: Lint.Utils.dedent`
+            \`${IGNORE_JSX_OPTION}\` specifies that jsx attributes with magic numbers should be ignored.
+            \`${ALLOWED_NUMBERS_OPTION}\` is a list of allowed numbers.`,
```

Comment:
Might need some help on this wording 🙃

Reply:
Yeah I've always thought the rule's docs were a little terse... how about:

```ts
Options may either be a list of numbers to ignore (not consider 'magic'), or an object containing up to two properties:

* \`${ALLOWED_NUMBERS_OPTION}\` as the list of numbers to ignore
* \`${IGNORE_JSX_OPTION}\` to specify that 'magic' numbers should be allowed as JSX attributes
```

`alignRule.ts` iis an example of formatting & phrasing here, though it doesn't have the two formats...

Do you have the time to add `optionExamples`, such as how `arrowParensRule.ts` has them? Those are particularly helpful for understanding rule options. If so, your test cases look pretty good as copy & paste fodder for them.